kentikapi/internal/api_resources/alerting.go

```diff
@@ -0,0 +1,74 @@
+package api_resources
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_connection"
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_endpoints"
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_payloads"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+type AlertingAPI struct {
+	BaseAPI
+}
+
+// NewAlertingAPI is constructor
+func NewAlertingAPI(transport api_connection.Transport) *AlertingAPI {
+	return &AlertingAPI{
+		BaseAPI{Transport: transport},
+	}
+}
+
+func (a *AlertingAPI) CreateManualMitigation(ctx context.Context, mm models.ManualMitigation) error {
+	payload := api_payloads.ManualMitigationToPayload(mm)
+	var response api_payloads.CreateManualMitigationResponse
+
+	if err := a.PostAndValidate(ctx, api_endpoints.ManualMitigationPath, payload, &response); err != nil {
+		return err
+	}
+
+	if response.Response.Result != "OK" {
+		return errors.New("Creating Manual Mitigation Failed")
+	}
+
+	return nil
+}
+
+func (a *AlertingAPI) GetActiveAlerts(ctx context.Context, startTime time.Time, endTime time.Time, filterBy string, filterVal string, showMitigations bool,
+	showAlarms bool, showMatches bool, learningMode bool) ([]models.Alarm, error) {
+	var response api_payloads.GetActiveAlertsResponse
+	timeFormatStr := "2006-01-02T15:04:05"
+	path := api_endpoints.GetActiveAlertsPath(startTime.Format(timeFormatStr), endTime.Format(timeFormatStr), filterBy, filterVal, boolToInt(showMitigations),
```

Comment:
Please call `startTime.Format(timeFormatStr)` and `boolToInt()` inside  api_endpoints.GetActiveAlertsPath - this is implementation detail of formatting the query string.

Reply:
Done